### PR TITLE
added missing method for parent resource container fallback

### PR DESCRIPTION
**Pull request by joehen**
this pr is adding a method for using the resource interfaces with parent resource containers that was missing from the simulated resource containers. This is needed for the test models for simulizar-5 to work

### CODE REVIEW

**Comment by joehen** [comment]
after two month someone should have a look at this basic pr...

File: bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/resources/SimulatedResourceContainer.java [hunk 1]
```diff
@@ -140,6 +140,41 @@ private IPassiveResource getSimplePassiveResource(final PassiveResource resource
...
+          } catch (final ResourceContainerIsMissingRequiredResourceType e) {
+              if (this.parentResourceContainer == null) {
+                  if (LOGGER.isEnabledFor(Level.ERROR)) {
+                      LOGGER.error("Resource container is missing a resource which was attempted to be loaded"
+                              + " by a component and has no parent Resource Container to look in. ID of resource type was: "
+                              + typeID);
+                  }
+                  throw new ResourceContainerIsMissingRequiredResourceType(typeID);
+              } else {
+                  this.parentResourceContainer.loadActiveResource(requestingProcess, resourceServiceID, typeID, demand);
```
Comment (sdkrach) [inline bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/resources/SimulatedResourceContainer.java:174]: The only thing that could be confusing is, that missing resource exceptions will further more only be thrown for the outer most resource container. Making it hard to trace back to the inner one, where the exception originally arose. We might be ok with that (seeing that its a special case anyhow). On the other hand if we imagine very many vms / pods inside a big host with particular local resources, the resulting exception would have very little information. 

We could simply catch this exception from the super implementation and rethrow the original one instead.

**Review by seiferma — CHANGES_REQUESTED** [review]
I cannot decide about the semantics of this change. However, it looks like all simulations, which worked before, will still work. Simulations failing previously might work now and it does not look like the new behavior would be unexpected.

Nevertheless, I have some change requests:
* please rebase your branch to the latest commit on master, so we can verify that the build is working
* unit tests of the changes would be nice (code coverage decreased)
* minor comments (see inline comments)

**Review by sdkrach — CHANGES_REQUESTED** [review]
@joehen you are right...
Straight forward implementation, I only have minor remarks

File: tests/de.uka.ipd.sdq.simucomframework.tests/src/de/uka/ipd/sdq/simucomframework/tests/SimulatedResourceContainerTests.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+		var foundContainer = nested.get(0);
+		assertEquals(childContainerId, foundContainer.getResourceContainerID(), "nested container should have our Id");
+		assertEquals(parentContainer, foundContainer.getParentResourceContainer(),
+				"child should have our parent as container");
+
+	}
+
+	@Test
+	void testNestedResourceContainerCalls() {
```
Comment (seiferma) [inline tests/de.uka.ipd.sdq.simucomframework.tests/src/de/uka/ipd/sdq/simucomframework/tests/SimulatedResourceContainerTests.java:92]: What does this test test? All calls for loading a resource are done on the parent container. I was wondering, what effect the child container actually has in this test case. Intuitively, I would think that the calls should be done on the child container to verify that the parent container is called.

**Review by seiferma — CHANGES_REQUESTED** [review]
Changes look good. However, I got a question about a test case.

File: tests/de.uka.ipd.sdq.simucomframework.tests/src/de/uka/ipd/sdq/simucomframework/tests/SimulatedResourceContainerTests.java (hunk 2 above)
Comment (joehen) [reply tests/de.uka.ipd.sdq.simucomframework.tests/src/de/uka/ipd/sdq/simucomframework/tests/SimulatedResourceContainerTests.java:92]: should be fixed, was just too much mockery :)

**Review by seiferma — APPROVED** [review]
Looks good to me now. Feel free to choose the merge approach you like ;)

**Pull request merged**